bin/fin

```diff
@@ -2855,6 +2856,68 @@ config_generate ()
 	fi
 }
 
+create_site ()
+{
+	while true; do
+		read -p "Please enter the name of your project (all one word, no spaces or slashes): " projectName
+		#todo, validate this input
```

Comment:
this precise syntax didn't work for me, but I've managed to fulfill the spirit of the request like so:

```bash
while true; do
  read -p "Please enter the name of your project (all one word, no spaces or slashes): " project_name
  if [[ "$project_name" =~ [^a-zA-Z0-9_-] ]]; then
    echo-red "Please provide a valid project name (only letters, numbers, dashes, and underscores are permitted)."
    continue
  fi
  break
done
```